testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -613,49 +618,97 @@ fn dump_helm_values_to_file(helm_release_name: &str, tmp_dir: &TempDir) -> Resul
     dump_string_to_file(file_name, content, tmp_dir)
 }
 
+struct K8sNamespacesApi {
+    api: Api<Namespace>,
+}
+
+#[async_trait]
+trait CreateNamespace: Send + Sync {
+    async fn create(&self, pp: &PostParams, namespace: &Namespace) -> Result<Namespace, KubeError>;
+}
+
+#[async_trait]
+impl CreateNamespace for Api<Namespace> {
+    async fn create(&self, pp: &PostParams, namespace: &Namespace) -> Result<Namespace, KubeError> {
+        self.create(pp, namespace).await
+    }
+}
+
+impl K8sNamespacesApi {
+    fn from_client(kube_client: K8sClient) -> Self {
+        K8sNamespacesApi {
+            api: Api::all(kube_client),
+        }
+    }
+}
+
+#[async_trait]
+impl CreateNamespace for K8sNamespacesApi {
+    async fn create(&self, pp: &PostParams, namespace: &Namespace) -> Result<Namespace, KubeError> {
+        self.api.create(pp, namespace).await
+    }
+}
+
+#[derive(Error, Debug)]
+#[error("{0}")]
+enum ApiError {
+    RetryableError(String),
+    FinalError(String),
+}
+
+async fn create_namespace(
+    namespace_creator: Arc<dyn CreateNamespace>,
+    kube_namespace: String,
+) -> Result<(), ApiError> {
+    let kube_namespace_name = kube_namespace.clone();
+    let namespace = Namespace {
+        metadata: ObjectMeta {
+            name: Some(kube_namespace_name.clone()),
+            ..ObjectMeta::default()
+        },
+        spec: None,
+        status: None,
+    };
+    if let Err(KubeError::Api(api_err)) = namespace_creator
```

Comment:
You only include api_err itself in one of the error arms, it might be good to do it in all of the arms for more info.